File: build_data.js
const fs = require('fs');

const freeContent = fs.readFileSync('C:/Users/user/Downloads/ATOMIC/atomic-habits-free-4 concepts-v3.md', 'utf-8');
const premiumContent = fs.readFileSync('C:/Users/user/Downloads/ATOMIC/the-habit-stress-test- PREMIUM.md', 'utf-8');

// Parse free concepts
const freeConceptsMatch = freeContent.split(/## Concept \\d+: /).slice(1);
const freeConcepts = freeConceptsMatch.map(c => {
    const titleMatch = c.match(/^(.*)\\r?\\n/);
    const title = titleMatch ? titleMatch[1].trim() : 'Unknown Concept';
    const markdown = c.substring(titleMatch ? titleMatch[0].length : 0).trim();
    return {
        title: title,
        markdown: markdown,
        isPremium: false
    };
});

// Parse premium concept
const premiumTitleMatch = premiumContent.match(/^# (.*)\\r?\\n/);
const premiumTitle = premiumTitleMatch ? premiumTitleMatch[1].trim() : 'The Habit Stress Test™';
const premiumMarkdown = premiumContent.replace(/^# .*\\r?\\n/, '').trim();

const premiumConcept = {
    title: premiumTitle,
    markdown: premiumMarkdown,
    isPremium: true
};

const allConcepts = [...freeConcepts, premiumConcept];

let dataJs = fs.readFileSync('data.js', 'utf-8');
const conceptsRegex = /concepts: \\[([\\s\\S]*?)\\]\\n\\s*}\\n\\s*\\]\\n\\s*\\}\\n\\];/;

const stringifiedConcepts = JSON.stringify(allConcepts, null, 4).replace(/\\n/g, '\\n                        ');
const newConceptsString = \concepts: \\\n            }\\n        ]\\n    }\\n];\;

dataJs = dataJs.replace(conceptsRegex, newConceptsString);
fs.writeFileSync('data.js', dataJs);
console.log('Successfully updated data.js');

File: build_data.py
import json
import re

with open('C:/Users/user/Downloads/ATOMIC/atomic-habits-free-4 concepts-v3.md', 'r', encoding='utf-8') as f:
    free_content = f.read()
    
with open('C:/Users/user/Downloads/ATOMIC/the-habit-stress-test- PREMIUM.md', 'r', encoding='utf-8') as f:
    premium_content = f.read()

# Parse free concepts
free_parts = re.split(r'## Concept \d+: ', free_content)[1:]
free_concepts = []
for part in free_parts:
    lines = part.split('\n', 1)
    title = lines[0].strip()
    markdown = lines[1].strip() if len(lines) > 1 else ''
    free_concepts.append({
        'title': title,
        'markdown': markdown,
        'isPremium': False
    })

# Parse premium
lines = premium_content.split('\n', 1)
premium_title = lines[0].replace('# ', '').strip()
premium_markdown = lines[1].strip() if len(lines) > 1 else ''
premium_concept = {
    'title': premium_title,
    'markdown': premium_markdown,
    'isPremium': True
}

all_concepts = free_concepts + [premium_concept]

with open('data.js', 'r', encoding='utf-8') as f:
    data_js = f.read()

# Very naive replacement of the concepts array inside atomic habits
import textwrap
concepts_str = json.dumps(all_concepts, indent=4)
indented_concepts = textwrap.indent(concepts_str, '                        ')

# We'll use regex to replace the concepts array
new_data_js = re.sub(
    r'concepts:\s*\[.*?\]\n\s*\}\n\s*\]\n\s*\}\n\];', 
    f'concepts: {indented_concepts.strip()}\n            }}\n        ]\n    }}\n];', 
    data_js, 
    flags=re.DOTALL
)

with open('data.js', 'w', encoding='utf-8') as f:
    f.write(new_data_js)

print("Updated data.js")
